File: example-android-project/app/src/main/java/io/github/gofaith/faithdroid/FViews/FFab.java
package io.github.gofaith.faithdroid.FViews;

import android.graphics.drawable.Drawable;
import android.os.Build;
import android.support.design.widget.FloatingActionButton;
import android.view.View;
import android.view.ViewGroup;

import faithdroid.Faithdroid;
import io.github.gofaith.faithdroid.UI.AttrGettable;
import io.github.gofaith.faithdroid.UI.AttrSettable;
import io.github.gofaith.faithdroid.UI.Toolkit;
import io.github.gofaith.faithdroid.UI.UIController;

public class FFab extends  FView implements AttrGettable, AttrSettable {
    public FloatingActionButton v;

    public FFab(UIController controller) {
        parentController=controller;
        v = new FloatingActionButton(parentController.activity);
        view=v;
        setElevation("8");
    }
    @Override
    public String getAttr(String attr) {
        String str = getUniversalAttr(attr);
        if (str != null) {
            return str;
        }
        switch (attr){
                //------------------
        }
        return "";
    }
    @Override
    public void setAttr(String attr, final String value) {
        if (setUniversalAttr(attr, value)) {
            return;
        }
        switch (attr){
            // -------------------------------------------------------------------
            case "Icon":
                Toolkit.file2Drawable(parentController, value, new Toolkit.OnDrawableReadyListener() {
                    @Override
                    public void onDrawableReady(Drawable drawable) {
                        v.setImageDrawable(drawable);
                    }
                });
                break;
            case "OnClick":
                v.setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        Faithdroid.triggerEventHandler(value, "");
                    }
                });
                break;
        }
    }
}

File: android/app/src/main/java/io/github/stevenzack/faithdroidapp/FViews/FFab.kt
package io.github.stevenzack.faithdroidapp.FViews

import android.graphics.drawable.Drawable
import android.view.View
import com.google.android.material.floatingactionbutton.FloatingActionButton
import faithdroid.Faithdroid
import io.github.stevenzack.faithdroidapp.UI.*

class FFab(controller: UIController) : FView(), AttrGettable, AttrSettable {
    var v: FloatingActionButton

    init {
        parentController = controller
        v = FloatingActionButton(parentController!!.activity)
        view = v
        setElevation("8")
    }

   override fun getAttr(attr: String): String {
        val str = getUniversalAttr(attr)
        if (str != null) {
            return str
        }
        when (attr) {

        }//------------------
        return ""
    }

   override fun setAttr(attr: String, value: String) {
        if (setUniversalAttr(attr, value)) {
            return
        }
        when (attr) {
            // -------------------------------------------------------------------
            "Icon" -> Toolkit.file2Drawable(parentController!!, value, {d->
                v.setImageDrawable(d)
            })
            "OnClick" -> v.setOnClickListener(object : View.OnClickListener {
                override fun onClick(v: View) {
                    Faithdroid.triggerEventHandler(value, "")
                }
            })
        }
    }
}
